src/views/dialogs/ShareAffiliateDialog.tsx

```diff
@@ -93,101 +95,116 @@ export const ShareAffiliateDialog = ({ setIsOpen }: DialogProps<ShareAffiliateDi
         },
       })}
     >
-      <div tw="column gap-1">
-        <div tw="row justify-between rounded-0.5 bg-color-layer-6 px-1 py-0.5">
-          <div>
-            <div tw="text-small text-color-text-0">
-              {data?.isEligible
-                ? stringGetter({ key: STRING_KEYS.AFFILIATE_LINK })
-                : stringGetter({
-                    key: STRING_KEYS.AFFILIATE_LINK_REQUIREMENT,
-                    params: {
-                      AMOUNT_USD: AFFILIATES_REQUIRED_VOLUME_USD.toLocaleString(),
-                    },
-                  })}
-            </div>
+      {!dydxAddress && (
+        <OnboardingTriggerButton
+          tw="w-full"
+          size={ButtonSize.Medium}
+          onClick={() => {
+            setIsOpen(false);
+          }}
+        />
+      )}
+      {dydxAddress && (
```

Comment:
this whole block remains the same, just wrapped around the `{dydxAddress && ..` condition